docs/debug_api.md

```diff
@@ -0,0 +1,165 @@
+Debug API
+=========
+
+
+The Debug API is a powerful HTTP API to make adjustments to the bridge at runtime, and is typically useful
+when administrating large bridges. Some functionality has moved to the [Admin Room](./admin_room.md)
+interface as sending commands over Matrix is preferred, however many powerful commands are exposed here.
+
+You can enable this feature in the config file:
+
+```yaml
+ircService:
+  # ...
+  debugApi:
+    enabled: true
+    port: 11100
+```
+
+**Note: The Debug API listens on `0.0.0.0` by default, so be careful to lock down access to this port.**
+
+To access the API over `curl`:
+
+```sh
+curl http://127.0.0.1:11100/inspectUsers?access_token=AS_TOKEN_FROM_REGISTRATION_FILE
+```
+
+## Endpoints
+
+<!-- no toc -->
+- [`GET /inspectUsers?regex={userRegex}`](#get-inspectusersregexuserregex)
+- [`POST /killRoom`](#post-killroom)
+- [`POST /killUser`](#post-killuser)
+- [`POST /reapUsers`](#post-reapusers)
+- [`GET /irc/$domain/user/$user_id`](#get-ircdomainuseruser_id)
+- [`POST /irc/$domain/user/$user_id`](#post-ircdomainuseruser_id)
+
+
+### `GET /inspectUsers?regex={userRegex}`
+
+#### Request Parameters
+
+- `userRegex` A JS regex string which should match against MXIDs. E.g. `@foobar_.*:matrix.org`
+
+#### Example Response
+
+```js
+{
+  "users": {
+    "@Half-Shot:half-shot.uk": [
+      {
+        "channels": [
+          "#matrix"
+        ],
+        "dead": false,
+        "server": "chat.freenode.net",
+        "nick": "Half-Shot"
+      }
+    ]
+  }
+}
+```
+
+### `POST /killRoom`
+
+Stop a room from being bridged. This will remove IRC ghost users from the room
+and disconnect Matrix users from the channel.
+
+The [Admin Room](admin_room.html#unlink) features a less
+powerful version of this command.
+
+#### Request Body
+
+```json5
+{
+  "room_id": "!foo:bar", // The Matrix Room ID. Required.
+  "domain": "irc.foo.bar", // The IRC domain. Required.
+  "channel": "#evilcave", // The IRC channel. Required.
+  "leave_notice": true, // Should a notice be sent on unbridge. Default: true.
+  "remove_alias": true, // Should the room alias for the room be removed. Default: true.
+}
+```
+
+#### Response Body
+
+The response body will contain a JSON array of stages that were successful and failed as
+it's possible for this command to only be partially successful.
+
+```json5
+error: [],
```

Comment:
Could we put some example errors and stages in there? From this sample it's hard to know what the JSON arrays will contain.